### ISSUE TITLE
Cannot set property data of # which has only a getter

### ISSUE THREAD

**Issue by andysay** [nuxt/framework#2768]
### Environment

------------------------------
- Operating System: `Darwin`
- Node Version:     `v16.13.0`
- Nuxt Version:     `3.0.0-27374259.21b664c`
- Package Manager:  `npm@8.1.0`
- Bundler:          `Vite`
- User Config:      `vue`, `privateRuntimeConfig`, `publicRuntimeConfig`, `css`, `build`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

-

### Describe the bug

Cannot set property data of # which has only a getter
at $fetchRaw2 (file://./node_modules/ohmyfetch/dist/chunks/fetch.mjs:142:25)
at processTicksAndRejections (node:internal/process/task_queues:96:5)
at async Module.useSettings (file://./.nuxt/dist/server/server.mjs:2947:18)
at async setup (file://./.nuxt/dist/server/server.mjs:2446:17)

**Comment by danielroe** [comment]
Would you provide a reproduction please?

**Comment by andysay** [comment]
i use composables 

```
export const useSettings = async () => {

    let settings =  await $fetch('/api/settings').then((res)  => {
        //@ts-ignore
        return res[0]
    })
    return settings
  }

```
 in components .vue file

```
  async setup() {
    let phone = await useSettings();
    return {
      phone,
    };

```

return Cannot set property data of # which has only a getter

**Comment by danielroe** [comment]
Working fine for me: https://stackblitz.com/edit/github-k1vsmk?file=app.vue. Try modifying that until you reproduce your error.

**Comment by andysay** [comment]
> Working fine for me: https://stackblitz.com/edit/github-k1vsmk?file=app.vue. Try modifying that until you reproduce your error.

@danielroe 

https://stackblitz.com/edit/github-k1vsmk-chcvm4?file=pages%2Findex.vue

**Comment by fmcdigital** [comment]
After updating to `3.0.0-27374259.21b664c` my `useFetch` stopped working and just returns a `null` value.

**Comment by misaon** [comment]
Same issue as @fmcdigital

**Comment by pi0** [comment]
Upstream issue solved in ohmyfetch@0.4.15. You can upgrade with `npx nuxi upgrade --force`
